src/adapter/src/catalog/consistency.rs

```diff
@@ -207,6 +218,61 @@ impl CatalogState {
                         }
                     }
                 }
+                CommentObjectId::Role(role_id) => {
+                    if self.roles_by_id.get(&role_id).is_none() {
+                        comment_inconsistencies
+                            .push(CommentInconsistency::Dangling(comment_object_id));
+                    }
+                }
+                CommentObjectId::Database(database_id) => {
+                    if self.database_by_id.get(&database_id).is_none() {
+                        comment_inconsistencies
+                            .push(CommentInconsistency::Dangling(comment_object_id));
+                    }
+                }
+                CommentObjectId::Schema((database, schema)) => {
+                    match (database, schema) {
+                        (
+                            ResolvedDatabaseSpecifier::Id(database_id),
+                            SchemaSpecifier::Id(schema_id),
+                        ) => {
+                            let schema = self
+                                .database_by_id
+                                .get(&database_id)
+                                .and_then(|database| database.schemas_by_id.get(&schema_id));
+                            if schema.is_none() {
+                                comment_inconsistencies
+                                    .push(CommentInconsistency::Dangling(comment_object_id));
+                            }
+                        }
+                        (ResolvedDatabaseSpecifier::Ambient, SchemaSpecifier::Id(schema_id)) => {
+                            if self.ambient_schemas_by_id.get(&schema_id).is_none() {
+                                comment_inconsistencies
+                                    .push(CommentInconsistency::Dangling(comment_object_id));
+                            }
+                        }
+                        // Temporary schemas are in the ambient database.
+                        (ResolvedDatabaseSpecifier::Id(_id), SchemaSpecifier::Temporary) => (),
```

Comment:
This should probably be some form of an error because temporary schemas should only be in the ambient DB (something for a later PR).